src/Graphics/Image/Processing/Dithering.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE BangPatterns #-} 
+
+-- | Adaptive histogram equalization is used to improve contrast in images.
+-- It adjusts image intensity in small regions (neighborhood) in the image.
+module Graphics.Image.Processing.Dithering where
+
+import Control.Monad (forM_, when)
+import Control.Monad.ST
+import Data.STRef 
+
+
+import Prelude as P hiding (subtract)
+import Graphics.Image.Processing.Filter
+import Graphics.Image.Interface as I
+import Graphics.Image
+import Graphics.Image.Types as IP
+import Graphics.Image.ColorSpace (X)
+
+
+applyThreshold :: RealFrac x => x -> x
+applyThreshold x = 255 * fromIntegral (floor(x/128))
+
+dithering
+  :: forall arr e cs . ( MArray arr RGB Double, IP.Array arr RGB Double, IP.Array arr RGB Double)
+  => Image arr RGB Double
+  -> Int  -- ^ width of output image
```

Comment:
You always ask for those arguments width and height, why is that? It's not C, you can call `dims` and get the dimensions of the source and result image

Reply:
@lehins Oh yes, very sorry for repeating the mistake. Wont happen again.